### ISSUE TITLE
[bug] ObjectLocation admin doesn't work in readonly mode

### ISSUE THREAD

**Issue by nemesifier** [openwisp/django-loci#95]
When a user has only view permission, the map doesn't work because the JS fails (it was not tested in this scenario).

**Comment by devkapilbansal** [comment]
> When a user has only view permission, the map doesn't work because the JS fails (it was not tested in this scenario).

I don't think it is due to the JS. If a user doesn't have change permissions, then these fields are not shown in ObjectLocation admin

I also found something in Django admin where we might found a possible fix :point_down: 
https://github.com/django/django/blob/cf79f92abee2ff5fd4fdcc1a124129a9773805b8/django/contrib/admin/options.py#L674

**Comment by pandafy** [comment]
> I also found something in Django admin where we might found a possible fix 
> https://github.com/django/django/blob/cf79f92abee2ff5fd4fdcc1a124129a9773805b8/django/contrib/admin/options.py#L674

That only affects ModelAdmin and not InlineAdmin. 

I created a [simple demo project](https://github.com/pandafy/inline_view_only) with our use case and also posted in [django-users mailing list](https://groups.google.com/g/django-users/c/3amTt7sPOD4) for help. 

Also, django-leaflet does not render map for view only permissions: https://github.com/makinacorpus/django-leaflet/issues/85

**Comment by DragnEmperor** [comment]
> > I also found something in Django admin where we might found a possible fix
> > https://github.com/django/django/blob/cf79f92abee2ff5fd4fdcc1a124129a9773805b8/django/contrib/admin/options.py#L674
> 
> That only affects ModelAdmin and not InlineAdmin.
> 
> I created a [simple demo project](https://github.com/pandafy/inline_view_only) with our use case and also posted in [django-users mailing list](https://groups.google.com/g/django-users/c/3amTt7sPOD4) for help.
> 
> Also, django-leaflet does not render map for view only permissions: [makinacorpus/django-leaflet#85](https://github.com/makinacorpus/django-leaflet/issues/85)

@pandafy i looked into it and it seems that for view only permissions, AdminReadonlyField is used and it looksup the fields in the instance, and hence does not uses the 'initial_data' which are used by `BoundFields`.
https://github.com/django/django/blob/ef62a3a68c9d558486145a42c0d71ea9a76add9e/django/contrib/admin/helpers.py#L265

I believe this can be solved by annotations or manually bypassing the permission checks to let BoundFields render and then setting the readonly and disabled attribute

**Comment by DragnEmperor** [comment]
Cool! Working on it.

**Comment by DragnEmperor** [comment]
@nemesifier I tried it locally and got the map to render, but i think it is rendering slowly. Shall I open a PR for this, will need your guidance if i am approaching this correctly or not.

**Comment by nemesifier** [comment]
> [@nemesifier](https://github.com/nemesifier) I tried it locally and got the map to render, but i think it is rendering slowly. Shall I open a PR for this, will need your guidance if i am approaching this correctly or not.

Great to ehar, looking forward to see it!
